src/seql/core.clj

```diff
@@ -295,7 +309,7 @@
                                                 :deserialize))
                     (map (process-compounds-fn (:schema env)
                                                qmeta))))
-          (recompose-relations fields)
+          (recompose-relations (:schema env) fields)
```

Comment:
The `schema` is required to infer the relation type, right?

Reply:
correct, schema is needed to identify the relation type. Will move it to the let as you recommend. thanks